pkg/destroy/azure/azure.go

```diff
@@ -226,9 +247,78 @@ func (o *ClusterUninstaller) Run() (*types.ClusterQuota, error) {
 		o.Logger.Debug(err)
 	}
 
+	if err := removeSharedTags(waitCtx, o.vnetClient, o.resourceGraphClient, o.InfraID, o.SubscriptionID); err != nil {
+		errs = append(errs, fmt.Errorf("failed to remove shared tags: %v", err))
+		o.Logger.Debug(err)
+	}
+
 	return nil, utilerrors.NewAggregate(errs)
 }
 
+func removeSharedTags(
+	ctx context.Context,
+	vnetClient *armnetwork.VirtualNetworksClient,
+	graphClient *armresourcegraph.Client,
+	infraID, subscriptionID string,
+) error {
+	tagKey := fmt.Sprintf("kubernetes.io_cluster.%s", infraID)
+	query := fmt.Sprintf(
+		"resources | where type == 'microsoft.network/virtualnetworks' and tags.['%s'] == 'shared' | project name, resourceGroup, tags",
+		tagKey,
+	)
+	results, err := graphClient.Resources(ctx,
+		armresourcegraph.QueryRequest{
+			Query: &query,
+			Subscriptions: []*string{
+				&subscriptionID,
+			},
+		},
+		nil,
+	)
+	if err != nil {
+		return fmt.Errorf("failed to query vnet: %w", err)
+	}
+
+	// marshal and unmarshal results.Data to avoid casting every type along the way.
+	// the sdk forcing us to do this is a shortcoming on its part which has been
+	// reported but so far no solution have been provided.
+	// see https://github.com/Azure/azure-sdk-for-go/issues/21004 for details.
+	b, err := json.Marshal(results.Data)
+	if err != nil {
+		return fmt.Errorf("failed to json marshall results data: %s", results.Data)
+	}
+	vnetsResult := []struct {
+		Name          string
+		ResourceGroup string
+		Tags          map[string]string
+	}{}
+	if err := json.Unmarshal(b, &vnetsResult); err != nil {
+		return fmt.Errorf("failed to json marshall results data: %s", results.Data)
+	}
+
+	// TODO: what do we do if there is more then one vnet with the shared tag?
```

Comment:
I need suggestions here - the query might return more than one vnet tagged with the `shared` tag. What should this do if that ever happens?

Reply:
Since we are destroying the cluster, I think it's fine/safe to remove all the `${clusterID} = shared` tags for all resources.

Reply:
done.